airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -3021,7 +3021,15 @@ def execute(self, context: Any):
         # Wait for the job to complete
         if not self.deferrable:
             job.result(timeout=self.result_timeout, retry=self.result_retry)
-            self._handle_job_error(job)
+            while True:
+                if job.state in ("PENDING", "RUNNING"):
+                    import time
+
+                    time.sleep(5)
+                    job.result(timeout=self.result_timeout, retry=self.result_retry)
```

Comment:
will calling `job.result` trigger a new job?

Reply:
`job.result` will not trigger a new job. `_submit_job` ([link](https://github.com/kandharvishnu/airflow/blob/fc52d7d127975097734217888c133923a9c4a5ec/providers/src/airflow/providers/google/cloud/operators/bigquery.py#L2621)) should trigger the job